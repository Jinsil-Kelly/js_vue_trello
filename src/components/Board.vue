<template>
  <div>
    <div>{{ bId }}</div>
    <!--<hr/>-->
    <!--<router-view></router-view>-->
  </div>
</template>

<script>
export default {
  created() {
    this.bId = this.$route.params.bId;
  },
  data() {
    return {
      bId: 0
    };
  },
  props: {
    msg: String
  }
};
</script>

<style lang="scss"></style>
