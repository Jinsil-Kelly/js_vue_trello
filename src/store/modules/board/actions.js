import * as api from "@/api/api";

export default {
  ADD_BOARD(context, { title }) {
    console.log(context);
    return api.board.create(title);
  },
  FETCH_BOARDS({ commit }) {
    return api.board.fetch().then(data => {
      commit("SET_BOARDS", data.list);
    });
  },
  FETCH_BOARD({ commit }, { id }) {
    return api.board.fetch(id).then(data => {
      commit("SET_BOARD", data.item);
    });
  }
};
