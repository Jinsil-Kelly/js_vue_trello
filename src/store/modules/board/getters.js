export default {
  boards: state => state.boards,
  board: state => state.board,
  isAddBoard: state => state.isAddBoard
};
