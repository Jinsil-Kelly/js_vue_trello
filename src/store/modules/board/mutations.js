export default {
  SET_IS_ADD_BOARD(state, toggle) {
    state.isAddBoard = toggle;
  },
  SET_BOARDS(state, boards) {
    state.boards = boards;
  }
};
