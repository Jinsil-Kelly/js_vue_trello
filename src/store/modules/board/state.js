export default {
  isAddBoard: false,
  boards: [],
  board: {},
  card: {}
};
