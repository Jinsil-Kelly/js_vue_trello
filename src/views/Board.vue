<template>
  <div v-if="loading">I am loading....</div>
  <div v-else>
    <div>{{ bId }}</div>

    <router-link :to="`/b/${bId}/c/1`">Card1</router-link>
    <router-link :to="`/b/${bId}/c/2`">Card2</router-link>
    <hr />
    <router-view></router-view>
  </div>
</template>

<script>
export default {
  data() {
    return { bId: 0, loading: true };
  },
  created() {
    return (this.bId = this.$route.params.bId);
  }
};
</script>
