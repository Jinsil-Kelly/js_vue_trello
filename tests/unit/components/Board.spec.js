import Board from "@/views/Board.vue";
import { shallowMount, createLocalVue, mount } from "@vue/test-utils";
import List from "@/components/List.vue";
import Vuex from "vuex";
import board from "../fixtures/board";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Board", () => {
  let wrapper, wrapperMounted;
  const getters = {
    board: jest.fn().mockReturnValue(board)
  };
  const actions = {
    FETCH_BOARD: jest.fn()
  };
  const mockStore = new Vuex.Store({
    modules: {
      board: {
        namespaced: true,
        actions,
        getters
      }
    }
  });
  const $route = {
    params: { bId: 2 }
  };
  const options = {
    store: mockStore,
    localVue,
    stubs: ["router-view"],
    mocks: {
      $route
    }
  };
  test("It should fetch items.", () => {
    wrapper = shallowMount(Board, options);

    expect(actions.FETCH_BOARD).toBeCalled();
    expect(actions.FETCH_BOARD.mock.calls[0][1]).toEqual({
      id: wrapper.vm.$route.params.bId
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    wrapperMounted = mount(Board, options);

    expect(wrapperMounted.find(".board").text()).toContain(board.title);
    expect(wrapperMounted.findAll(List).length).toEqual(3);
    expect(wrapperMounted.find(List).exists()).toBe(true);
    expect(wrapperMounted.html()).toMatchSnapshot();
  });
});
