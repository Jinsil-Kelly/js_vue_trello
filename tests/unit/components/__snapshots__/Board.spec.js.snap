// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board It should fetch items. 1`] = `
<div>
  <div class="board-wrapper">
    <div class="board">
      <div class="board-header"><span class="board-title">MockTitleForTest</span></div>
      <div class="list-section-wrapper">
        <div class="list-section">
          <div class="list-wrapper">
            <list-stub data="[object Object]"></list-stub>
          </div>
          <div class="list-wrapper">
            <list-stub data="[object Object]"></list-stub>
          </div>
          <div class="list-wrapper">
            <list-stub data="[object Object]"></list-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
  <router-view-stub></router-view-stub>
</div>
`;

exports[`Board renders main child components 1`] = `
<div>
  <div class="board-wrapper">
    <div class="board">
      <div class="board-header"><span class="board-title">MockTitleForTest</span></div>
      <div class="list-section-wrapper">
        <div class="list-section">
          <div class="list-wrapper">
            <div class="list">
              <div class="list-header">
                <div class="list-header-title">Todo</div>
              </div>
              <div data-list-id="1" class="card-list"></div>
              <div><a class="add-card-btn">
                  + Add a card...
                </a></div>
            </div>
          </div>
          <div class="list-wrapper">
            <div class="list">
              <div class="list-header">
                <div class="list-header-title">Doing</div>
              </div>
              <div data-list-id="2" class="card-list"></div>
              <div><a class="add-card-btn">
                  + Add a card...
                </a></div>
            </div>
          </div>
          <div class="list-wrapper">
            <div class="list">
              <div class="list-header">
                <div class="list-header-title">Done</div>
              </div>
              <div data-list-id="3" class="card-list"></div>
              <div><a class="add-card-btn">
                  + Add a card...
                </a></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <router-view-stub></router-view-stub>
</div>
`;
